planning/behavior_path_planner/include/behavior_path_planner/scene_module/avoidance/avoidance_utils.hpp

```diff
@@ -29,8 +29,25 @@ using behavior_path_planner::PlannerData;
 struct PolygonPoint
 {
   geometry_msgs::msg::Point point;
+  size_t bound_seg_idx;
+  double lon_dist_to_segment;
   double lat_dist_to_bound;
-  double lon_dist;
+
+  bool is_after(const PolygonPoint & other_point) const
+  {
+    if (bound_seg_idx == other_point.bound_seg_idx) {
+      return other_point.lon_dist_to_segment < lon_dist_to_segment;
+    }
+    return other_point.bound_seg_idx < bound_seg_idx;
+  }
+
+  bool is_outside_bounds(const bool is_on_right) const
+  {
+    if (is_on_right) {
+      return lat_dist_to_bound < 0.0;
+    }
+    return 0.0 < lat_dist_to_bound;
+  };
 };
 
```

Comment:
if the changes is not so much, I would suggest use `using PolygonPoints = std::vector<PolygonPoint>` to make some function argument shorter etc. but since this might be just a preference, it is okay to ignore.

Reply:
Hmm, I prefer `std::vector<PolygonPoint>` since we know the detailed type only by the type name.